src/main.rs

```diff
@@ -62,6 +66,16 @@ mod tests {
     /// An integration test for the `run` command.
     #[test]
     fn test_handle_run_command() {
-        handle_run_command(&get_model_dir());
+        handle_run_command(&get_model_dir()).unwrap();
+
+        // Second time will fail because the logging is already initialised
+        assert_eq!(
+            handle_run_command(&get_model_dir())
+                .unwrap_err()
+                .chain()
+                .next()
+                .map(|x| format!("{x}")),
+            Some("Failed to initialize logging.".to_owned())
+        );
```

Comment:
Would inverting the commands work? No idea, just thinking on alternatives:
```suggestion
        handle_run_command(&get_model_dir())
            .unwrap_or_else(|err| assert_eq!(err, Some("Failed to initialize logging.".to_owned())));
```
Or something along these lines, using the same `unwrap_or_else` to check that the error message is what it should rather than printing it.

I guess that if there's no error, this will pass, which is not what we want...

Reply:
You'd still need the `err.chain().next().map(...)` to get the first context string and convert it into a String. And, like you said, if there is not error it will pass, which is not what we want.

Reply:
I found a slightly more readable way to do it